himawari_readwrite.f90

```diff
@@ -504,19 +506,20 @@ integer function AHI_readchan(fname, indata,band,convert,ahi_nav,verbose)result(
 	real 						::	temp,temp2
 	integer					::	arrxs,arrys,filelun,flen
 	real(kind=ahi_dreal)	::	gain,offset,c0,c1,c2,lspd,plnk,bolz,clamb
-
+        logical:: ef
+ 
 	bval	=	band
 
 	if (band.eq.1.or.band.eq.2.or.band.eq.4) then
-		open(newunit=filelun, file=fname,form='unformatted',action='read',status='old',access='stream',convert='little_endian')
+		open(unit=filelun, file=fname,form='unformatted',action='read',status='old',access='stream',convert='little_endian')
```

Comment:
I'm a bit confused by this, `newunit` is Fortran 2008 syntax so should work well on any new compiler! Newunit is the recommended choice as it negates the possibility of using an already-open file handler.